File: Hiking/Unity/Assets/JumpTo.js
#pragma strict

var player : GameObject;
var PlayerScript : PlayerScript;
var mySelf : GameObject;
var isMouseOver : boolean;
var newPosition : Vector3;
var speed: float;

function Start () {
//player = GameObject.Find("Player");
mySelf = gameObject;
PlayerScript = player.GetComponent.<PlayerScript>();

}

function Update () 
{
//gameObject.GetComponent(ParticleSystem).enabled = true;
if(this.isMouseOver == true && Input.GetMouseButtonDown(0))
	{
	PlayerScript.jumpTo(mySelf);
	
//	iTween.MoveBy(player, iTween.Hash("y", lobHeight, "time", lobTime/2, "easeType", iTween.EaseType.easeOutQuad));
//	iTween.MoveBy(player, iTween.Hash("y", -lobHeight, "time", lobTime/2, "delay", lobTime/2, "easeType", iTween.EaseType.easeInCubic));     
//	iTween.MoveTo(player, iTween.Hash("position", mySelf.transform.position, "time", lobTime, "easeType", iTween.EaseType.linear));

	
	
	}
	
	
if(Input.GetMouseButtonDown(1)){
PlayerScript.jT();
}

}// update

    function OnCollisionEnter() {
   // if(collision.gameObject.tag=="Player"){
    Debug.Log("hit");
   // }
    }

function OnMouseEnter () {
//mySelf.light.enabled = true;
isMouseOver = true;

}

function OnMouseExit () {
//mySelf.light.enabled = false;
isMouseOver = false;

}

File: Hiking/Unity/Assets/JumpTo.cs
using UnityEngine;
using System.Collections;



public class JumpTo : MonoBehaviour {

	
public GameObject player; //the player
public bool isMouseOver = false; //is mouse over? (sometimes you don't need a comment I guess!)
public PlayerScript PlayerScript; //the player's scripting.
public GameObject mySelf;
	
	
	// Use this for initialization
	void Start () {
		mySelf = this.gameObject;
	PlayerScript = player.GetComponent<PlayerScript>();
		
	}
	
	// Update is called once per frame
	void Update () {
	
		if(isMouseOver == true && Input.GetMouseButtonDown(0)) //if player clicks on me
			{

			PlayerScript.JumpTo(gameObject.transform); //call the player's JumpTo() to make them jump to me
	
			}
	
		
		
	}
	
	void OnMouseEnter () {
//mySelf.light.enabled = true;
isMouseOver = true;
		}

void OnMouseExit () {
//mySelf.light.enabled = false;
isMouseOver = false;

}
}

